components/core/View.tsx

```diff
@@ -0,0 +1,173 @@
+    readonly borderTopWidth?: RNViewStyle['borderTopWidth'],
+    readonly borderWidth?: RNViewStyle['borderWidth'],
+    readonly borderColor?: RNViewStyle['borderColor'],
+
+    readonly display?: RNViewStyle['display'],
+    readonly overflow?: RNViewStyle['overflow'],
+  ) {}
+}
+
+class ViewAliasProps {
+  constructor(
+    readonly bg?: RNViewStyle['backgroundColor'],
+    readonly bgColor?: RNViewStyle['backgroundColor'],
+
+    readonly p?: RNViewStyle['padding'],
+    readonly px?: RNViewStyle['paddingHorizontal'],
+    readonly ph?: RNViewStyle['paddingVertical'],
+    readonly pt?: RNViewStyle['paddingTop'],
+    readonly pl?: RNViewStyle['paddingLeft'],
+    readonly pr?: RNViewStyle['paddingRight'],
+    readonly pb?: RNViewStyle['paddingBottom'],
+
+    readonly m?: RNViewStyle['margin'],
+    readonly mx?: RNViewStyle['marginHorizontal'],
+    readonly my?: RNViewStyle['marginVertical'],
+    readonly mt?: RNViewStyle['marginTop'],
+    readonly ml?: RNViewStyle['marginLeft'],
+    readonly mr?: RNViewStyle['marginRight'],
+    readonly mb?: RNViewStyle['marginBottom'],
+  ) {}
+}
+
+type ViewStyleProp = keyof ViewStyleProps | keyof ViewAliasProps;
+const viewStylePropKeys: ViewStyleProp[] = (Object.keys(new ViewStyleProps()) as ViewStyleProp[]).concat(Object.keys(new ViewAliasProps()) as ViewStyleProp[]);
+
+const propAliasMapping: Record<keyof ViewAliasProps, keyof ViewStyleProps> = {
+  bg: 'backgroundColor',
+  bgColor: 'backgroundColor',
+
+  p: 'padding',
+  px: 'paddingHorizontal',
+  ph: 'paddingVertical',
+  pt: 'paddingTop',
+  pl: 'paddingLeft',
+  pr: 'paddingRight',
+  pb: 'paddingBottom',
+
+  m: 'margin',
+  mx: 'marginHorizontal',
+  my: 'marginVertical',
+  mt: 'marginTop',
+  ml: 'marginLeft',
+  mr: 'marginRight',
+  mb: 'marginBottom',
+} as const;
+
+const dimensionalProps: ViewStyleProp[] = [
+  'bottom',
+  'flexBasis',
+  'height',
+  'left',
+  'margin',
+  'marginBottom',
+  'marginHorizontal',
+  'marginLeft',
+  'marginRight',
+  'marginTop',
+  'marginVertical',
+  'maxHeight',
+  'maxWidth',
+  'minHeight',
+  'minWidth',
+  'padding',
+  'paddingBottom',
+  'paddingHorizontal',
+  'paddingLeft',
+  'paddingRight',
+  'paddingTop',
+  'paddingVertical',
+  'right',
+  'top',
+  'width',
+];
+
+const validateProp = (prop: ViewStyleProp, value): void => {
+  if (dimensionalProps.includes(prop) && typeof value === 'string') {
+    // Dimensions as strings must either specify 'pt' or '%'
+    if (value.slice(-2) !== 'pt' && value.slice(-1) !== '%') {
+      const error = `Invalid string value "${value}" for property ${prop}: string dimensions must specify pt or %`;
+      console.error(error);
+      throw new Error(error);
+    }
+  }
+};
+
+export interface ViewProps extends RNViewProps, ViewStyleProps, ViewAliasProps {}
+export const View: React.FC<ViewProps> = ({children, style = {}, ...props}) => {
+  const resolvedProps: RNViewProps = {style};
```

Comment:
the logic in this method is slightly different from the way I implemented the TextWrapper - in this component, all top-level style properties are assumed to overwrite any `style` entries that are set.

generally I would be suspicious of code that's setting the same property through `style` and through top-level props - we should avoid that unless there's an amazing reason to do so.

I'll do a revision on Text to make its behavior the same as this method.

Reply:
Either here or as a follow-up, this method sounds like a good candidate for a couple unit tests to document & lock down the expected behavior.

Reply:
fixed in #75